tcms/report/views.py

```diff
@@ -502,8 +502,7 @@ def _report_context(self):
             else:
                 errors = form.errors
 
-        queries = fmt_queries(queries)
-        del queries['report type']
+        del queries['report_type']
```

Comment:
Here queries can still be None and this will raise an exception. Try testing manually the report page to discover when it will fail.

Reply:
There are more hidden surprises here which are only revealed when you perform more thorough testing.